python/runtime/common/py_NoiseModel.cpp

```diff
@@ -37,10 +37,41 @@ void bindNoiseModel(py::module &mod) {
   mod.def("unset_noise", &unset_noise,
           "Clear backend simulation from any existing noise models.");
 
-  py::class_<kraus_op>(mod, "KrausOperator", py::buffer_protocol(),
-                       "The KrausOperator is represented by a matrix and "
-                       "serves as an element of a "
-                       "quantum channel such that Sum Ki Ki^dag = I.")
+  py::class_<noise_model>(
+      mod, "NoiseModel",
+      "The `NoiseModel` defines a set of :class:`KrausChannel`'s applied to "
+      "specific qubits after the invocation of specified quantum operations.")
+      .def(py::init<>(), "Construct an empty noise model.")
+      .def(
+          "add_channel",
+          [](noise_model &self, std::string &opName,
+             std::vector<std::size_t> &qubits, kraus_channel &channel) {
+            self.add_channel(opName, qubits, channel);
+          },
+          py::arg("operator"), py::arg("qubits"), py::arg("channel"),
+          R"#(Add the given :class:`KrausChannel` to be applied after invocation 
+of the specified quantum operation.
+
+Args:
+  operator (str): The quantum operator to apply the noise channel to.
+  qubits (List[int]): The qubit/s to apply the noise channel to.
+  channel (cudaq.KrausChannel): The :class:`KrausChannel` to apply 
+    to the specified `operator` on the specified `qubits`.)#")
+      .def(
+          "get_channels",
+          [](noise_model self, const std::string &op,
+             std::vector<std::size_t> qubits) {
```

Comment:
Could we use reference types here to prevent unnecessary copies:  (`noise_model& self, const std::string &op, const std::vector<std::size_t>& qubits)`?